File: app/src/main/java/com/mcwilliams/theninjamethod/strava/authorize/AccessScope.java
package com.mcwilliams.theninjamethod.strava.authorize;

public enum AccessScope {
    PUBLIC("public"),
    WRITE("write"),
    VIEW_PRIVATE("view_private"),
    VIEW_PRIVATE_WRITE("activity:write,read");

    private String rawValue;

    AccessScope(String rawValue) {
        this.rawValue = rawValue;
    }
    
    @Override
    public String toString() {
        return rawValue;
    }
}

File: app/src/main/java/com/mcwilliams/theninjamethod/strava/AccessScope.kt
package com.mcwilliams.theninjamethod.strava

enum class AccessScope(private val rawValue: String) {
    PUBLIC("public"),
    WRITE("write"),
    VIEW_PRIVATE("view_private"), VIEW_PRIVATE_WRITE("activity:read,activity:write");

    override fun toString(): String {
        return rawValue
    }

}